Adding Disqus Notification Emails

gep13 opened the pull request:
Attempts to implement requirements for #27
- Added new MessageService method to construct email to current maintainers
- Added same method signature to interface
- Created new ViewModel for sending id and text of comment from the client
- Added new PackagesController method for accepting post from client when new Disqus comment is added
- Added new Route to handle mapping to new controller method

gep13 commented:
@ferventcoder this likely still needs some work/refactoring, but it is at least at a point that it does "something" so thought I would open it for review.

Be gentle :smile_cat:

ferventcoder commented:
(GH-27) is what you are looking for in the commits. :)

ferventcoder commented:
Nevermind, looks like that works as well. :)

gep13 commented:
> (GH-27) is what you are looking for in the commits. :)

Dang it! I forgot that you use that format.  I will fix them up so that they are consistent.

ferventcoder commented on chocolatey/Website/Services/MessageService.cs:
```diff
@@ -148,6 +148,49 @@ public void SendContactOwnersMessage(MailAddress fromAddress, PackageRegistratio
+    To stop receiving contact emails as a maintainer of this package, sign in to the {2} and 
+    [change your email notification settings]({3}).
+</em>";
+            string subject = "[{0}] Message for maintainers of the package '{1}'";
```
[{0}] New disqus comment for maintainers of '{1}'

gep13 replied:
Done, good catch. Over zealous copy and paste! :+1:

ferventcoder commented on chocolatey/Website/Services/MessageService.cs:
```diff
@@ -148,6 +148,49 @@ public void SendContactOwnersMessage(MailAddress fromAddress, PackageRegistratio
+<em style=""font-size: 0.8em;"">
+    To stop receiving contact emails as a maintainer of this package, sign in to the {2} and 
+    [change your email notification settings]({3}).
+</em>";
```
You can leave this (162-166) off. If they turn off email, they have no way to receive notifications otherwise.

ferventcoder replied:
From some of the design, I think it was originally intended that folks would get notifications in the application.

ferventcoder commented on chocolatey/Website/Services/MessageService.cs:
```diff
@@ -148,6 +148,49 @@ public void SendContactOwnersMessage(MailAddress fromAddress, PackageRegistratio
+
+        public void SendCommentNotificationToMaintainers(PackageRegistration packageRegistration, CommentViewModel comment, string emailSettingsUrl, string packageUrl)
+        {
+            string body = @"Possible Action Required: A comment has been added to the Disqus Forum Thread for Package {0}.  As a maintainer this package, please read this comment, and take appropriate actions, if required.  The comment that was left was:'._
```
I would shift this up a bit 

Comment: {0}

This comment has been added to the disqus forum thread for {1}. It may not show up immediately if it is subject to moderation but we wanted you to know about it.

ferventcoder commented:
Looks pretty good.

gep13 commented on chocolatey/Website/Services/MessageService.cs:
```diff
@@ -148,6 +148,49 @@ public void SendContactOwnersMessage(MailAddress fromAddress, PackageRegistratio
+            {
+                mailMessage.Subject = subject;
+                mailMessage.Body = body;
+                mailMessage.From = new MailAddress("whatshouldthisbe@idontknow.com");
```
@ferventcoder did you see this?  I mentioned it in my email.  What should the from address be?  It threw an exception if I didn't set it to something.

ferventcoder replied:
I meant to comment on that. Note the others use an email address.

ferventcoder replied:
`MailAddress fromAddress` is passed in.

gep13 replied:
@ferventcoder Yes, I saw that, but in this case, the user can be anyone, not necessarily an authenticated user, so I can't pass in an email address, unless there is a default one that I can use.

ferventcoder replied:
use the default one.

gep13 replied:
which is :-) The chocolatey webadmin group email?

gep13 replied:
Never mind, found it:

```mailMessage.From = new MailAddress(settings.GalleryOwnerEmail, settings.GalleryOwnerName);```

gep13 commented:
Based on comments, the email body has been trimmed down as follows:

![image](https://cloud.githubusercontent.com/assets/1271146/5603812/ac253938-938e-11e4-81d4-9a58e3a933f6.png)

gep13 commented:
@ferventcoder I have just rebased all the commits into one, and updated comment.  Let me know what you think about the last outstanding comment, and I can fix that up to. Thanks

gep13 commented:
@ferventcoder ok, last comment has now been taken care of.  Let me know if there is anything else, or whether we can get this deployed to staging for more testing.

ferventcoder commented:
I thought I saw an earlier picture somewhere where you had mentioned the moderation possibility. I remember liking your wording better.

ferventcoder commented:
It looks good. I can deploy it to test but we'd need that disqus configurability. If you want to add that as part of this, that would be fine. But I could understand breaking it up as a separate PR.

gep13 commented:
> I thought I saw an earlier picture somewhere where you had mentioned the moderation possibility. I remember liking your wording better.

Dang it, I thought you wanted it simpler, i.e. as per your code comment.  I will see if I can re-create the wording, as it is now lost due to rebase.

> It looks good. I can deploy it to test but we'd need that disqus configurability. If you want to add that as part of this, that would be fine. But I could understand breaking it up as a separate PR.

Away to sign off for tonight, but I will think about this, most likely be a different PR.  I would need a heads up about how to do the configuration in AppHarbor, as I have never done anything like that.  Would it just be an AppSettings in Web.config file, and then added to the MVC View?

ferventcoder commented:
Never lost

ferventcoder commented:
yes on web.config appsettings.

gep13 commented:
Where would I find them? I thought rebase rewrites the history?

Sent from my Windows Phone
________________________________
From: Rob Reynolds<mailto:notifications@github.com>
Sent: ‎03/‎01/‎2015 22:55
To: chocolatey/chocolatey.org<mailto:chocolatey.org@noreply.github.com>
Cc: Gary Ewan Park<mailto:gep13@gep13.co.uk>
Subject: Re: [chocolatey.org] Adding Disqus Notification Emails (#140)

Never lost


---
Reply to this email directly or view it on GitHub:
https://github.com/chocolatey/chocolatey.org/pull/140#issuecomment-68612935

ferventcoder commented:
It does, but it keeps the old commits around for at least a little while. You can inspect the diff of an old commit with git reflog

ferventcoder commented:
This is good enough to pull. I will go ahead with it.

ferventcoder commented:
Ah shoot. Can you rebase this?

gep13 commented:
> Ah shoot. Can you rebase this?

Done.

ferventcoder commented:
You are the man! I was about to add your remote and give it a shot.

gep13 commented on chocolatey/Website/Views/Packages/DisplayPackage.cshtml:
```diff
@@ -487,6 +487,7 @@
     @{
         var disqusUrl = "http://" + Request.Url.Host + @Url.Package(Model.Id);
```
@ferventcoder this http:// needs to be made https://
